content/applications/productivity/discuss/advanced/email_servers.rst

```diff
@@ -224,6 +224,20 @@ alias in your mail server.
 
 .. _Office 365 documentation:
     https://support.office.com/en-us/article/how-to-set-up-a-multifunction-device-or-application-to-send-email-using-office-365-69f58e99-c550-4274-ad18-c805d654b4c4
+    
+Set up different outgoing servers in a multi-company environment
+================================================================
+
+The "From Filter" allows to use a specific outgoing depending of the "From". If an email is sent from Odoo, the email server will be chosen as follows:
+
+#.  Odoo searches for an email server having the same "From Filter" as the "From" defined in the outgoing email. For example, if the "From" is foo@bar.com, only the email servers having the "From Filter" equals to foo@bar.com will be returned.
+
+#. If not found, Odoo searches for an email server having the same domain in the "From Filter" as in the "From". For example, if the "From" is foo@bar.com, only the email servers having the "From Filter" equals to bar.com will be returned.
+
+#. If not found, Odoo will return all email servers without any "From Filter".
```

Comment:
@std-odoo  Can you review the following edit, did I understand it correctly? The attached edit replaces this sentence: 
``If no email servers are found after checking for the domain, then Odoo will return all email servers without any "From Filter" set.`` Can you clarify?

What happens if there aren't any results from this query? Does anything need to be added about the notification encapsulation? Should we add this as a note to the other two pull requests on the from filter? Thanks :)

``Should this previous query return no results, then Odoo performs a search for an email server using the system parameter: 'mail.default.from'. First, the email address listed will attempt to match an email server and then the domain will attempt a match.``

Reply:
You have some steps missing I think, to resume the entire flow
1. Find a mail server whose from_filter match the entire sender email address
2. If not found, find a mail server whose from_filter match the sender email domain name
3. If not found, find a mail server with `from_filter=False` (match everything)
4. If not found find a mail server whose from_filter match notifications@odoo.com
5. If not found, find a mail server whose from_filter match the domain name of the "notifications email"
6. If not found return the first outgoing mail server (sorted by priority), in that case the mail configuration is bad and the email might me marked as spam

> If no email servers are found after checking for the domain, then Odoo will return all email servers without any "From Filter" set. 

So this is correct for step 3

> Should this previous query return no results, then Odoo performs a search for an email server using the system parameter: 'mail.default.from'. First, the email address listed will attempt to match an email server and then the domain will attempt a match

This is correct for step 4 and 5

Hope it clarify

Reply:
you can check in the server log to be sure the email was sent, but if you have at least one server and if it didn't match the email / notification email, yes it might be marked as spam.